pkg/v1/cli/discovery/kubernetes_test.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package discovery_test
+
+import (
+	"errors"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/vmware-tanzu/tanzu-framework/apis/cli/v1alpha1"
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli/discovery"
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli/plugin"
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/fakes"
+	fakehelper "github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/fakes/helper"
+)
+
+func TestClient(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Discovery Suite")
+}
+
+var _ = Describe("Unit tests for kubernetes discovery", func() {
+	var (
+		err                  error
+		currentClusterClient *fakes.ClusterClient
+		kd                   *discovery.KubernetesDiscovery
+		plugins              []plugin.Discovered
+		cliplugins           []v1alpha1.CLIPlugin
+	)
+
+	Describe("When Getting Discovered plugins from k8s cluster", func() {
```

Comment:
nit-ish: In all the tests, wouldn't it be more logical to have the `Expect(err).NotTo(HaveOccurred())` as the first check before the other checks?